lib/blockstore/file.js

```diff
@@ -0,0 +1,590 @@
+/*!
+ * blockstore/file.js - file blockstore for bcoin
+ * Copyright (c) 2019, Braydon Fuller (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const {isAbsolute, resolve, join} = require('path');
+const bdb = require('bdb');
+const assert = require('bsert');
+const fs = require('bfile');
+const bio = require('bufio');
+const hash256 = require('bcrypto/lib/hash256');
+const Network = require('../protocol/network');
+const AbstractBlockStore = require('./abstract');
+const {BlockRecord, FileRecord} = require('./records');
+const layout = require('./layout');
+const {types, prefixes} = require('./common');
+
+/**
+ * File Block Store
+ *
+ * @alias module:blockstore:FileBlockStore
+ * @abstract
+ */
+
+class FileBlockStore extends AbstractBlockStore {
+  /**
+   * Create a blockstore that stores blocks in files.
+   * @constructor
+   */
+
+  constructor(options) {
+    super(options);
+
+    assert(isAbsolute(options.location), 'Location not absolute.');
+
+    this.location = options.location;
+    this.indexLocation = resolve(this.location, './index');
+
+    this.db = bdb.create({
```

Comment:
Are you purposely disallowing configuration of the different `DBOptions` found here? https://github.com/bcoin-org/bdb/blob/b8a48714edfc443d4e0cff2eba803f60e0e891b9/lib/db.js#L1261

Reply:
This is done as `location` is `this.indexLocation` and does not directly correspond with `options` and to also be explicit to which options are passed through.